lib/ansible/module_utils/docker/common.py

```diff
@@ -565,10 +565,21 @@ def find_image(self, name, tag):
             # In API <= 1.20 seeing 'docker.io/<name>' as the name of images pulled from docker hub
             registry, repo_name = auth.resolve_repository_name(name)
             if registry == 'docker.io':
-                # the name does not contain a registry, so let's see if docker.io works
-                lookup = "docker.io/%s" % name
-                self.log("Check for docker.io image: %s" % lookup)
-                images = self._image_lookup(lookup, tag)
+                # If docker.io is explicitly there in name, the image
+                # isn't found in some cases (#41509)
+                self.log("Check for docker.io image: %s" % repo_name)
+                images = self._image_lookup(repo_name, tag)
+                if len(images) == 0 and repo_name.startswith('library/'):
```

Comment:
Is their anyway we could end up looking at the wrong image?

Reply:
I double-checked this; it turns out one cannot overwrite the default registry in docker (moby/moby#7203), so this cannot happen.

(Except if someone replaces docker daemon by some "compatible" thing which does it differently. But then, there are probably no guarantees about anything...)